### DISCUSSION TITLE
Sorting table containing components, not just text

### DISCUSSION THREAD

**Discussion by caleyshemc** [TanStack/table#2056] [General]
I'm trying to use `useGlobalFilter` to sort a table containing a mix of text and `<Link>`s. However, the filtering does not work on any row value that is wrapped in a Link. For example:

```
<Link to='/example'>
    Example Link
</Link>
```

Here's a filterType I've made to get the Link text:
```
const filterTypes = {
  text: (rows, id, filterValue) => {
    return rows.filter(row => {
      const rowValue = row.values[id];
      return rowValue !== undefined
        ? rowValue.props.children
            .toLowerCase()
            .startsWith(String(filterValue).toLowerCase())
        : true;
    });
  }
};
```

`rowValue.props.children` does find "Example Link", but this doesn't work for columns with plain text instead of a Link. Is there an easier way to do this? Could I perhaps pass a "filter value" option somewhere, and filter by that?

**Comment by Ajayghoshm** [comment]
Is there any solution for this?
Facing the same issue.
